### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: src/amici.cpp [hunk 1]
```diff
@@ -64,6 +64,19 @@ void runAmiciSimulation(UserData *udata, const ExpData *edata, ReturnData *rdata
     applyChainRuleFactorToSimulationResults(udata, rdata, edata);
 
 freturn:
+    if(*pstatus<0){
+        if(llhdata)
+            *llhdata = amiGetNaN();
+        
+        if(sllhdata)
+            for (int ip=0; ip<nplist; ip++)
+            sllhdata[ip] = amiGetNaN();
```
Comment (dweindl) [inline src/amici.cpp]: fillArray(sllhdata, nplist, amiGetNaN());

fillArray(s2llhdata, nplist*ng, amiGetNaN());

would save some space

Comment (dweindl) [inline src/amici.cpp]: or put it to a separate function 
void invalidateReturnData(ReturnData*, UserData*);

File: src/amici_interface_matlabo2.cpp [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "amici_interface_matlab.cpp"
```
Comment (dweindl) [inline src/amici_interface_matlabo2.cpp]: What is this file required for?

Comment (FFroehlich) [inline src/amici_interface_matlabo2.cpp]: this is necessary to have debug symbols for both first and secondorder compilation, otherwise object file will be overwritten.

Comment (dweindl) [inline src/amici_interface_matlabo2.cpp]: Hm.. but why is it even compiled twice? The object file should be usable for both versions.
